lxd/container_lxc.go

```diff
@@ -8233,6 +8322,29 @@ func (c *containerLXC) restartProxyDevices() error {
 func (c *containerLXC) createNetworkDevice(name string, m types.Device) (string, error) {
 	var dev, n1 string
 
+	d, err := device.New("ct", device.Config(m))
```

Comment:
So I think we could greatly simplify this logic by having a function like `func (c *containerLXC) deviceCreate(hotplug bool, m types.Device)` which would then deal with all those steps.

Per our call today, I think we also want a function on the device struct to check hotplugabilitiy so we don't need to start special casing things like phys.

Reply:
@stgraber yes that sounds like good plan.

Reply:
@stgraber have added these now.